vpr/src/base/vpr_context.h

```diff
@@ -287,7 +287,9 @@ struct RoutingContext : public Context {
 
     std::vector<t_rr_node_route_inf> rr_node_route_inf; /* [0..device_ctx.num_rr_nodes-1] */
 
-    std::vector<unsigned int> non_configurable_bitmap; /*[0...device_ctx.num_rr_nodes/32] */
+    //Information about whether a node is a non-configurable node 0: configurable, 1: non-configurable
+    //Initialized once when RoutingContext is initialized, static throughout invocation of router 
+    std::vector<bool> non_configurable_bitmap; /*[0...device_ctx.num_rr_nodes-1] */
```

Comment:
`std::vector<bool>` should not be used, as it is terribly inefficent because it attempts to conform to the `std::vector` interface.  Suggested implementation is over here: https://github.com/verilog-to-routing/vtr-verilog-to-routing/blob/d64dce7bb5082d82ede91fe4065c229cbc8f2a29/libs/libvtrutil/src/vtr_dynamic_bitset.h

Reply:
Hi Keith, I have already gathered data for an implementation similar to the suggested implementation you have. Since the bitset implementation you have is not checked in yet, should I integrate your vtr_dynamic_bitset.h into my change or should I just put my bitset implementation into master branch and wait to be replaced later on?

Reply:
vtr_dynamic_bitset.h has been merged, so you should be able to use it now.